File: abc181-200/abc194/e/main.py
N,M = map(int,input().split())
A = list(map(int,input().split()))
count = [0] * (N + 1)

for i in range(M):
	count[A[i]] += 1

for i in range(M + 1):
	if count[i] == 0:
		mex = i
		break

for i in range(N - M):
	count[A[i]] -= 1
	count[A[i + M]] += 1
	if count[A[i]] == 0:
		mex = min(mex, A[i])

print(mex)

File: abc181-200/abc194/e/main.rb
N,M,*A=$<.read.split.map &:to_i
x=[0]*-~N
A[...M].each{|a|x[a]+=1}
m=x.index(0)
(0...N-M).each{|i|x[A[i+M]]+=1
if 0==x[A[i]]-=1 then;m=[m,A[i]].min;end}
p m
